Framework/API/src/MatrixWorkspace.cpp

```diff
@@ -477,18 +477,26 @@ bool MatrixWorkspace::hasGroupedDetectors() const {
  *    KEY is the DetectorID (pixel ID)
  *    VALUE is the Workspace Index
  *  @param throwIfMultipleDets :: set to true to make the algorithm throw an
- * error
- *         if there is more than one detector for a specific workspace index.
+ * error if there is more than one detector for a specific workspace index.
+ *  @param ignoreIfNoValidDets :: set to true to exclude spectra that do not
+ * include any valid detector IDs. Note that if any valid detector IDs exist
+ * then all of the detector IDs (including the invalid ones) will be returned.
  *  @throw runtime_error if there is more than one detector per spectrum (if
  * throwIfMultipleDets is true)
  *  @return Index to Index Map object. THE CALLER TAKES OWNERSHIP OF THE MAP AND
  * IS RESPONSIBLE FOR ITS DELETION.
  */
-detid2index_map MatrixWorkspace::getDetectorIDToWorkspaceIndexMap(bool throwIfMultipleDets) const {
+detid2index_map MatrixWorkspace::getDetectorIDToWorkspaceIndexMap(bool throwIfMultipleDets,
+                                                                  bool ignoreIfNoValidDets) const {
   detid2index_map map;
+  auto specInfo = spectrumInfo();
```

Comment:
Just a quick question. Did we mean to copy the `SpectrumInfo` object here rather than just by const ref?

Reply:
Oops, no - thanks for spotting that!